internal/provider/job_resource_test.go

```diff
@@ -0,0 +1,622 @@
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+)
+
+func TestParseHttpResponse(t *testing.T) {
+	templateID := basetypes.NewInt64Value(1)
+	inventoryID := basetypes.NewInt64Value(2)
+	extraVars := jsontypes.NewNormalizedNull()
+	testTable := []struct {
+		name     string
+		body     []byte
+		expected jobResourceModel
+		failure  bool
+	}{
+		{
+			name:    "no ignored fields",
+			failure: false,
+			body:    []byte(`{"job_type": "run", "url": "/api/v2/jobs/14/", "status": "pending"}`),
+			expected: jobResourceModel{
+				TemplateID:    templateID,
+				Type:          types.StringValue("run"),
+				URL:           types.StringValue("/api/v2/jobs/14/"),
+				Status:        types.StringValue("pending"),
+				InventoryID:   inventoryID,
+				ExtraVars:     extraVars,
+				IgnoredFields: types.ListNull(types.StringType),
+			},
+		},
+		{
+			name:    "ignored fields",
+			failure: false,
+			body: []byte(`{"job_type": "run", "url": "/api/v2/jobs/14/", "status": 
+			"pending", "ignored_fields": {"extra_vars": "{\"bucket_state\":\"absent\"}"}}`),
+			expected: jobResourceModel{
+				TemplateID:    templateID,
+				Type:          types.StringValue("run"),
+				URL:           types.StringValue("/api/v2/jobs/14/"),
+				Status:        types.StringValue("pending"),
+				InventoryID:   inventoryID,
+				ExtraVars:     extraVars,
+				IgnoredFields: basetypes.NewListValueMust(types.StringType, []attr.Value{types.StringValue("extra_vars")}),
+			},
+		},
+		{
+			name:     "bad json",
+			failure:  true,
+			body:     []byte(`{job_type: run}`),
+			expected: jobResourceModel{},
+		},
+	}
+	for _, tc := range testTable {
+		t.Run(tc.name, func(t *testing.T) {
+			d := jobResourceModel{
+				TemplateID:  templateID,
+				InventoryID: inventoryID,
+				ExtraVars:   extraVars,
+			}
+			err := d.ParseHTTPResponse(tc.body)
+			if tc.failure {
+				if err == nil {
+					t.Errorf("expecting failure while the process has not failed")
+				}
+			} else {
+				if err != nil {
+					t.Errorf("unexpected process failure (%s)", err.Error())
+				} else if !reflect.DeepEqual(tc.expected, d) {
+					t.Errorf("expected (%v) - result (%v)", tc.expected, d)
+				}
+			}
+		})
+	}
+}
+
+// DeepEqualJSONByte compares the JSON in two byte slices.
+func DeepEqualJSONByte(a, b []byte) (bool, error) {
+	var j1, j2 interface{}
+	if err := json.Unmarshal(a, &j1); err != nil {
+		return false, err
+	}
+	if err := json.Unmarshal(b, &j2); err != nil {
+		return false, err
+	}
+	return reflect.DeepEqual(j2, j1), nil
+}
+
+func TestCreateRequestBody(t *testing.T) {
+	testTable := []struct {
+		name     string
+		input    jobResourceModel
+		expected []byte
+	}{
+		{
+			name: "unknown fields",
```

Comment:
What is the difference between this test and [this one](https://github.com/ansible/terraform-provider-aap/pull/2/files#diff-9783f57a592d7669379e472320b72831a01eec44cf40760d6a519357cfee72afR120) ?

Reply:
@GomathiselviS the link is pointing the #diff of the PR, please specify the file and the line or the test case name

Reply:
Sorry about it. I am ltrying to understand the difference between 2 testcases - https://github.com/ansible/terraform-provider-aap/blob/68b6369c3b314fed66279ac93dfe85f5bf1116af/internal/provider/job_resource_test.go#L112 and https://github.com/ansible/terraform-provider-aap/blob/68b6369c3b314fed66279ac93dfe85f5bf1116af/internal/provider/job_resource_test.go#L120

Reply:
@GomathiselviS the Terraform `attr.value` can be `null` (empty value) or `unknown` (unset), these 2 test cases are different in the sense that they are testing these different cases